src/broker/modservice.c

```diff
@@ -209,7 +211,7 @@ static int register_event (modservice_ctx_t *ctx, const char *name,
     rc = 0;
 cleanup:
     flux_msg_handler_destroy (mh);
-    free (match.topic_glob);
+    free ((char*)match.topic_glob);
```

Comment:
Difficult to tell just looking at the diff, but could the cast to `(char *)` become unnecessary if you `free (topic)` instead of `match.topic_glob`? (Applies to the other cases in this PR as well)